lib/puppetserver/ca/action/list.rb

```diff
@@ -79,16 +89,55 @@ def run(input)
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
 
-          (all || certnames.any?) \
-            ? output_certs_by_state(requested, signed, revoked, missing)
-            : output_certs_by_state(requested)
+          (all || certnames.any?)\
+          ? output_certs_by_state(all, output_format, requested, signed, revoked, missing)
+          : output_certs_by_state(all, output_format, requested)
 
           return missing.any? \
             ? 1
             : 0
         end
+        
+        def output_certs_by_state(all, output_format, requested, signed = [], revoked = [], missing = [])
+          if output_format == 'json'
+            output_certs_json_format(all, requested, signed, revoked, missing)
+          else
+            output_certs_text_format(requested, signed, revoked, missing)
+          end
+        end
+
+        def output_certs_json_format(all, requested, signed, revoked, missing)
+          grouped_cert = {}
+
+          if all
+            grouped_cert = { "requested" => requested,
+                             "signed" => signed,
+                             "revoked" => revoked }.to_json
+            @logger.inform(grouped_cert)
+          else
+            unless requested.empty?
+              grouped_cert["requested"] = requested
+            end
+
+            unless signed.empty?
+              grouped_cert["signed"] = signed
+            end
+
+            unless revoked.empty?
+              grouped_cert["revoked"] = revoked
+            end
+
+            unless missing.empty?
+              grouped_cert["missing"] = missing
+            end
+
+            grouped_cert.empty? \
+            ? @logger.inform({ "requested" => requested }.to_json) 
```

Comment:
It'd be nice to at least have a comment here about how we should only get here if neither `--all` or `--certname` was passed and so we will just be giving a list of the CSRs.